Kotlin Continuations should be excluded from requestBody

_Cross-posting this from:_ https://github.com/quarkusio/quarkus/issues/21509

This appears to relate to #893. It should be possible to fix by filtering `kotlin.coroutines.Continuation` from `requestBody`. I'm happy to create a PR, if someone can point me in the right direction. :)

I _think_ I'm on the right track, but I'm unable to run the tests due to a confusing error.

```sh
$ ./mvnw test -f extensions/smallrye-openapi/deployment/ -Dtest=OpenApiPathWithSegmentsTestCase
...
Caused by: java.lang.ClassCastException: class io.quarkus.test.common.http.TestHTTPConfigSourceProvider cannot be cast to class org.eclipse.microprofile.config.spi.ConfigSourceProvider (io.quarkus.test.common.http.TestHTTPConfigSourceProvider is in unnamed module of loader 'app'; org.eclipse.microprofile.config.spi.ConfigSourceProvider is in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @5faeeb56)
        at io.quarkus.runtime.configuration.RuntimeConfigSourceProvider.getConfigSources(RuntimeConfigSourceProvider.java:20)
        at io.quarkus.runtime.configuration.ConfigUtils.addSourceProvider(ConfigUtils.java:192)
        at io.quarkus.runtime.generated.Config.readConfig(Config.zig:1012)
        at io.quarkus.deployment.steps.RuntimeConfigSetup.deploy(RuntimeConfigSetup.zig:42)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:261)
        ... 49 more

```

https://github.com/rgmz/quarkus/tree/fix/smallrye-openapi/kotlin-continuations

--- a/core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AbstractAnnotationScanner.java
+++ b/core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AbstractAnnotationScanner.java
@@ -1,13 +1,14 @@
 package io.smallrye.openapi.runtime.scanner.spi;
 
-import io.smallrye.openapi.api.OpenApiConfig;
-import io.smallrye.openapi.api.constants.OpenApiConstants;
-import org.eclipse.microprofile.openapi.models.Extensible;
-
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
+
+import org.eclipse.microprofile.openapi.models.Extensible;
+
+import io.smallrye.openapi.api.OpenApiConfig;
+import io.smallrye.openapi.api.constants.OpenApiConstants;
 
 /**
  * Abstract base class for annotation scanners

--- a/core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AnnotationScanner.java
+++ b/core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AnnotationScanner.java
@@ -13,6 +13,7 @@ import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
+import java.util.stream.IntStream;
 
 import org.eclipse.microprofile.openapi.models.Components;
 import org.eclipse.microprofile.openapi.models.OpenAPI;
@@ -906,8 +907,9 @@ public interface AnnotationScanner {
     }
 
     /**
-     * Go through the method parameters looking for one that is not annotated with a jax-rs/spring
-     * annotation.That will be the one that is the request body.
+     * Go through the method parameters looking for one that is not a Kotlin Continuation,
+     * is not annotated with a jax-rs/spring annotation, and is not a known path parameter.
+     * That will be the one that is the request body.
      * 
      * @param context the scanning context
      * @param method MethodInfo
@@ -916,22 +918,19 @@ public interface AnnotationScanner {
      */
     default Type getRequestBodyParameterClassType(final AnnotationScannerContext context, MethodInfo method,
             final ResourceParameters params) {
-        List<AnnotationScannerExtension> extensions = context.getExtensions();
+
         List<Type> methodParams = method.parameters();
-        if (methodParams.isEmpty()) {
-            return null;
-        }
-        for (short i = 0; i < methodParams.size(); i++) {
-            List<AnnotationInstance> parameterAnnotations = JandexUtil.getParameterAnnotations(method, i);
-            if (parameterAnnotations.isEmpty()
-                    || !containsScannerAnnotations(parameterAnnotations, extensions)) {
-                // Also check if there is a PathPatameter already with this
-                if (!isPathParameter(context, method.parameterName(i), params)) {
-                    return methodParams.get(i);
-                }
-            }
-        }
-        return null;
+
+        return IntStream.range(0, methodParams.size())
+                .filter(position -> !isKotlinContinuation(methodParams.get(position)))
+                .filter(position -> !isPathParameter(context, method.parameterName(position), params))
+                .filter(position -> {
+                    List<AnnotationInstance> annotations = JandexUtil.getParameterAnnotations(method, (short) position);
+                    return annotations.isEmpty() || !containsScannerAnnotations(annotations, context.getExtensions());
+                })
+                .mapToObj(methodParams::get)
+                .findFirst()
+                .orElse(null);
     }
 
     default boolean isPathParameter(final AnnotationScannerContext context, String name, final ResourceParameters params) {

--- a/core/src/test/java/io/smallrye/openapi/runtime/scanner/spi/AbstractAnnotationScannerTest.java
+++ b/core/src/test/java/io/smallrye/openapi/runtime/scanner/spi/AbstractAnnotationScannerTest.java
@@ -1,16 +1,17 @@
 package io.smallrye.openapi.runtime.scanner.spi;
 
-import io.smallrye.openapi.api.OpenApiConfig;
-import io.smallrye.openapi.api.models.OperationImpl;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.util.Collections;
 import java.util.Set;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.smallrye.openapi.api.OpenApiConfig;
+import io.smallrye.openapi.api.models.OperationImpl;
 
 class AbstractAnnotationScannerTest {
     /**

--- a/extension-jaxrs/src/test/java/io/smallrye/openapi/runtime/scanner/ApiResponseTests.java
+++ b/extension-jaxrs/src/test/java/io/smallrye/openapi/runtime/scanner/ApiResponseTests.java
@@ -242,6 +242,7 @@ class ApiResponseTests extends IndexScannerTestBase {
     @Test
     void testKotlinContinuationStringResponse() throws IOException, JSONException {
         @jakarta.ws.rs.Path("/hello")
+        @jakarta.ws.rs.Consumes({ "text/plain" })
         class Resource {
 
             /**

